### PR TITLE
Feature - Weapon list layout draft

### PR DESCRIPTION
**Pull request by sol3uk**
Solution to #28 
Introduces the concept of view "modes" to the weapon screen, whilst simultaneously meeting the criteria for #28, where challenges are shown as text.
If we wanted this to be more compact there are a few options, however any more compact and it risks going against the initial style/design of the site.

I also removed the word "Get" from the start of all challenges (This can be reverted, just helped with text space)

Thoughts @carlssonemil?

P.S. I can put together another version with the 3 mastery camos grouped up at the bottom, but then there's the issue with not seeing the mastery challenge text in plain text (some categories vary each challenge inside them)

### CODE REVIEW

File: src/components/WeaponComponent.vue [hunk 1]
```diff
@@ -18,22 +18,26 @@
 			v-if="!weapon.comingSoon"
 			class="progress"
 			:style="{
-				'grid-template-columns': `repeat(${Object.keys(camouflages).length}, 1fr)`,
+				'grid-template-rows': `${ layout === 'list' ?  'repeat(' + Object.keys(camouflages).length + ', 1fr)' : ''}`,
+				'grid-template-columns': `${layout === 'grid' ? 'repeat(' + Object.keys(camouflages).length + ', 1fr)' : ''}`,
```
Comment (sol3uk) [inline src/components/WeaponComponent.vue]: Is there a better way to do this with Vue?

Comment (carlssonemil) [reply src/components/WeaponComponent.vue]: Not really, you could define the entire style binding as an object in the script tag to make the HTML more readable, [more information here](https://vuejs.org/guide/essentials/class-and-style.html#binding-inline-styles), but personally I like to do it this way, inline.

File: src/components/WeaponComponent.vue [hunk 2]
```diff
@@ -156,21 +174,21 @@ export default {
 
 	.name {
 		$padding: 25px;
+		$list-padding: 10px;
```
Comment (carlssonemil) [inline src/components/WeaponComponent.vue]: Could have a bit more padding here still 😄 
```suggestion
		$list-padding: 20px 10px;
```

File: src/components/WeaponComponent.vue [hunk 3]
```diff
@@ -156,21 +174,21 @@ export default {
...
 		justify-content: center;
-		padding: $padding 0;
 		position: relative;
 		transition: $transition;
 		user-select: none;
 
 		@media (max-width: $tablet) {
 			font-size: 18px;
-			padding: #{$padding * 2} $padding;
+			padding: $list-padding $padding;
```
Comment (carlssonemil) [inline src/components/WeaponComponent.vue]:
```suggestion
			padding: $list-padding;
```

File: src/components/WeaponComponent.vue [hunk 4]
```diff
@@ -187,6 +205,17 @@ export default {
 			@include polyatomic-camouflage-background;
 			color: white;
 		}
+
+		&.grid {
+			padding: $padding 0;
+		}
+
+		&.list {
+			padding: $list-padding 0;
```
Comment (carlssonemil) [inline src/components/WeaponComponent.vue]:
```suggestion
			padding: $list-padding;
```

File: src/components/WeaponComponent.vue [hunk 5]
```diff
@@ -198,6 +227,85 @@ export default {
...
+					padding: 8px;
+
+					.name {
+						font-size: 14px;
+					}
+				}
+			}
+
+			&.weapon-layout-list > .inner {
+				$image-size: 30px;
```
Comment (carlssonemil) [inline src/components/WeaponComponent.vue]: We can slightly decrease the camouflage-image size
```suggestion
				$image-size: 25px;
```

File: src/components/WeaponComponent.vue [hunk 6]
```diff
@@ -198,6 +227,85 @@ export default {
...
+					z-index: 2;
+				}
+
+				.info {
+					padding: 0 5px;
+					text-align: left;
+
+					.name {
+						font-weight: 500;
+					}
```
Comment (carlssonemil) [inline src/components/WeaponComponent.vue]: This is probably not needed, right? 🤔 
```suggestion

```

File: src/components/WeaponComponent.vue [hunk 7]
```diff
@@ -277,14 +385,6 @@ export default {
...
-					height: 100%;
-					object-fit: cover;
-					position: relative;
-					width: 100%;
-					z-index: 1;
-				}
-
 				p {
 					font-size: 14px;
 				}
```
Comment (carlssonemil) [inline src/components/WeaponComponent.vue]: I think this can be removed
```suggestion

```

File: src/components/WeaponComponent.vue [hunk 8]
```diff
@@ -198,6 +227,85 @@ export default {
...
+					opacity: 0;
+					position: absolute;
+					transform: translate(-50%, -50%);
+					transition: $transition;
+					top: 50%;
+					z-index: 2;
+				}
+
+				.info {
+					padding: 0 5px;
```
Comment (carlssonemil) [inline src/components/WeaponComponent.vue]: Let's increase the padding a bit here and add y-axis padding for the requirements that span two rows (sniper rifles for example)
```suggestion
					padding: 4px 8px;
```

File: src/components/WeaponComponent.vue [hunk 9]
```diff
@@ -198,6 +227,85 @@ export default {
...
+					text-align: left;
+
+					.name {
+						font-weight: 500;
+					}
+
+					.requirement {
+						display: block;
+						font-size: 11px;
+						line-height: 1.5;
```
Comment (carlssonemil) [inline src/components/WeaponComponent.vue]: Let's decrease the line-height a bit here
```suggestion
						line-height: 1.2;
```

**Review by carlssonemil — CHANGES_REQUESTED** [review]
Nice work! 🥳 Got a few minor CSS suggestions! 😄 

Could probably also slightly increase the requirement text on mobile, for easier readability 🤔

**Review by carlssonemil — APPROVED** [review]
Nice! 👏

**Pull request merged**